shared/src/main/scala/squants/Quantity.scala

```diff
@@ -308,8 +308,3 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
   def map(f: Double ⇒ Double): A = unit(f(value))
 }
 
-abstract class StrictlyPositiveQuantity[A <: StrictlyPositiveQuantity[A]](value: Double) extends Quantity[A] {self: A ⇒
```

Comment:
Sorry about this. I think this can be made to work just making the constructors return `Option` or `Either` or even `Try`